core/src/display_object/stage.rs

```diff
@@ -365,6 +379,7 @@ impl<'gc> Stage<'gc> {
     ) {
         if display_state == self.display_state()
             || (Self::is_fullscreen_state(display_state) && self.is_fullscreen())
+            || !self.allow_full_screen()
```

Comment:
I think we should check for `is_fullscreen_state(display_state)` (probably move into the expression above)

Reply:
Are you sure you can change the display state to Normal when `allowFullScreen` is not true in Adobe Flash Player? <https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display/Stage.html#displayState> says a SecurityError is thrown when trying to set the display state "if the param or embed HTML tag's allowFullScreen attribute is not set to true", mentioning nothing about that only applying when setting it to a fullscreen state.

Reply:
I haven't tested it myself.

Reply:
Okay, I don't know either.

Reply:
Ah, just read a bit more: "Full-screen mode is initiated in response to a mouse click or key press by the user; the movie cannot change `Stage.displayState` without user input." Since on web no scripts can make the SWF start in full screen without input, letting you change the `displayState` to normal would do nothing since the `displayState` can never start as anything else and can't be changed to full screen.

Reply:
I did `window.addEventListener("click", () => document.querySelector("embed, object").requestFullscreen())` in Pale Moon on http://new.weedtowonder.org/domestication.html. The content's full screen and I can use escape to exit full screen, but the real question is, did the `displayState` of the stage change? I don't have a good test SWF for this so I don't know. Ideally I'd have an SWF with a button that displays the current displayState when clicked, but I don't have Adobe Animate so making tests is a PITA.

Reply:
[displayStateAvm1.swf.zip](https://github.com/ruffle-rs/ruffle/files/13994943/displayStateAvm1.swf.zip)

Tested this and I can't seem to cause the displayState to change without using AS, so I think my logic here is fine.